packagedb/api.py

```diff
@@ -778,10 +783,17 @@ def get_all_versions(purl: PackageURL):
     if not package_name or not versionAPI:
         return
 
-    all_versions = versionAPI().fetch(package_name)
+    all_versions = versionAPI().fetch(package_name) or []
     versionClass = VERSION_CLASS_BY_PACKAGE_TYPE.get(purl.type)
 
-    return [versionClass(package_version.value) for package_version in all_versions]
+    result = []
+    for package_version in all_versions:
+        try:
```

Comment:
Like above, it seems heavy-handed to pass exceptions like this. 

@pombredanne Should we log the invalid package versions instead?

Reply:
Do we have a real life data example that would fail?

Reply:
@pombredanne @JonoYang I don't have a real-life example at hand, but all the version classes in the universe raise an `InvalidVersion` error when fail to parse the input version. If these exceptions are not handled, the consumer side will receive a 500 internal server error. IMO since we don't have control over the input version we should handle these errors and log them. I will add logging for both exception handling.